stubs/regex/regex/regex.pyi

```diff
@@ -0,0 +1,349 @@
+def sub(
+    pattern: AnyStr,
+    repl: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def sub(
+    pattern: _regex.Pattern,
+    repl: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def subf(
+    pattern: AnyStr,
+    format: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def subf(
+    pattern: _regex.Pattern,
+    format: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def subn(
+    pattern: AnyStr,
+    repl: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def subn(
+    pattern: _regex.Pattern,
+    repl: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def subfn(
+    pattern: AnyStr,
+    format: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> tuple[AnyStr, int]: ...
+@overload
+def subfn(
+    pattern: _regex.Pattern,
+    format: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> tuple[AnyStr, int]: ...
+
+Pattern: _regex.Pattern
+Match: _regex.Match
```

Comment:
This throws an error in the stubtest as they are types generated from the Pattern object rather than the object itself: https://github.com/mrabarnett/mrab-regex/blob/79d633b2dc9899e0f0cbb9ffdb81c14a4f7a86f2/regex_3/regex.py#L722-L723

Not sure how to handle this? (would be handy if I could use these types withing the stubs)